Dungeoneer/src/com/interrupt/dungeoneer/gfx/GlRenderer.java

```diff
@@ -667,9 +666,8 @@ private void drawUI() {
 		uiBatch.setColor(Color.WHITE);
 
 		if(OverlayManager.instance.current() == null || !OverlayManager.instance.current().catchInput) {
-			// show crosshair if a ranged weapon is held
-			Item held = game.player.GetHeldItem();
-			if (held != null && (held.itemType == ItemType.bow || held.itemType == ItemType.junk || held.itemType == ItemType.wand)) {
+			// Draw crosshair
+			if (showCrosshair()) {
 				if(!Options.instance.hideUI) {
```

Comment:
What about moving this check for `Options.instance.hideUI` also into this new method `showCrosshair`?

It could be used as an early bail-out. When we do not want to render the UI, why bother to first check if we have the correct item held?

Reply:
Makes sense to me.